frontend/src/app/main/data/workspace/tokens/application.cljs

```diff
@@ -284,21 +285,71 @@
    (when (number? value)
      (generate-text-shape-update {:letter-spacing (str value)} shape-ids page-id))))
 
+(defn warn-font-variant-not-found! []
+  (st/emit!
+   (ntf/show {:content (tr "workspace.tokens.font-variant-not-found")
+              :type :toast
+              :level :warning
+              :timeout 7000})))
+
+(defn- update-closest-font-variant-id-by-weight
+  [txt-attrs target-variant font-id on-mismatch]
+  (let [font (fonts/get-font-data font-id)
+        variant (when font
+                  (fonts/find-closest-variant font (:weight target-variant) (:style target-variant)))
+        call-on-mismatch? (when (and (fn? on-mismatch) variant)
+                            (or
+                             (not= (:font-weight target-variant) (:weight variant))
+                             (when (:font-style target-variant)
+                               (not= (:font-style target-variant) (:style variant)))))]
+    (when call-on-mismatch?
+      (on-mismatch))
+    (cond-> txt-attrs
+      (:id variant) (assoc :font-variant-id (:id variant)))))
+
+(defn- generate-font-family-text-shape-update
+  [txt-attrs shape-ids page-id on-mismatch]
+  (let [not-found-font (= (:font-id txt-attrs) (str uuid/zero))
+        update-node? (fn [node]
+                       (or (txt/is-text-node? node)
+                           (txt/is-paragraph-node? node)))
+        update-fn (fn [node find-closest-weight?]
+                    (let [font-id (if not-found-font (:font-id node) (:font-id txt-attrs))
+                          txt-attrs (cond-> txt-attrs
+                                      find-closest-weight? (update-closest-font-variant-id-by-weight node font-id on-mismatch))]
+                      (-> node
+                          (d/txt-merge txt-attrs)
+                          (cty/remove-typography-from-node))))]
+    (dwsh/update-shapes shape-ids
+                        (fn [shape]
+                          (txt/update-text-content shape update-node? #(update-fn %1 (ctst/font-weight-applied? shape)) nil))
+                        {:ignore-touched true
+                         :page-id page-id})))
+
+(defn- create-font-family-text-attrs
+  [value]
+  (let [font-family (-> (first value)
+                        ;; Strip quotes around font-family like `"Inter"`
```

Comment:
Is this not done by the function that parses the font family string into an array?